chain/syncer.go

```diff
@@ -478,22 +431,22 @@ func (syncer *Syncer) HandleNewTipset(ctx context.Context, tsKey types.TipSetKey
 	return nil
 }
 
-func (syncer *Syncer) exceedsFinalityLimit(chain []types.TipSet) (bool, error) {
-	if len(chain) == 0 {
-		return false, nil
+func (syncer *Syncer) exceedsFinalityLimit(newHead types.TipSet) (bool, error) {
```

Comment:
There are 2 errors this method can return:
1. Calls to Tipset methods that return the undefined tipset error
2. The single call to `GetTipSet` that will return an error if chainStore is unable to find a block. A method on the ChainStore like `GetHeadTipSet` that doesn't error would be a "nice to have" here since it seems unlikely that we will fail to find the tipset for our current head.

This method is only used in the fetchers call back, and it would be great if we could remove the error from its return.

Reply:
I think it makes sense that the callback should be capable of propagating an error.